app/controllers/api/v1/labels.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require 'doorkeeper/grape/helpers'
+
+module API
+  module V1
+    class Labels < Grape::API
+      helpers Doorkeeper::Grape::Helpers
+
+      before { doorkeeper_authorize! }
+
+      before do
+        def current_account
+          Account.find(doorkeeper_token.resource_owner_id)
+        end
+      end
+
+      resource :labels do
+        desc 'List all labels for current account.'
+        get do
+          present current_account.labels, with: API::Entities::Label
+        end
+
+        desc 'Return a label by key.'
+        params do
+          requires :key, type: String, allow_blank: false, desc: 'Label key.'
+        end
+        route_param :key do
+          get do
+            label = Label.find_by(key: params[:key])
+            if label.present?
+              present label, with: API::Entities::Label
+            else
+              error!('Couldn\'t find Label', 404)
+            end
+          end
+        end
+
+        desc "Create a label with 'general' scope."
+        params do
+          requires :key, type: String, allow_blank: false, desc: 'Label key.'
+          requires :value, type: String, allow_blank: false, desc: 'Label value.'
+        end
+        post do
+          label =
+            current_account.labels.new(
+              key: params[:key],
+              value: params[:value],
+              scope: 'general'
+            )
+          if label.save
+            present label, with: API::Entities::Label
+          else
+            error!(label.errors.as_json(full_messages: true), 422)
+          end
+        end
+
+        desc 'Update a label.'
+        params do
+          requires :key, type: String, allow_blank: false, desc: 'Label key.'
+          requires :value, type: String, allow_blank: false, desc: 'Label value.'
+        end
+        patch ':key' do
+          label = current_account.labels.find_by(key: params[:key])
+          return error!('Couldn\'t find Label', 404) if label.blank?
+
+          if label.general?
```

Comment:
I would block those routes at an earlier stage we would not authorize Updates / Delete for scope private.

Reply:
`before` blocks don't understand filtering by action name, so done this in the action's code